Modules/FindCUDA/select_compute_arch.cmake

```diff
@@ -0,0 +1,175 @@
+# BorIFrowed from caffe
+# https://github.com/BVLC/caffe/blob/master/cmake/Cuda.cmake
+
+# Synopsis:
+# Function for selecting GPU arch flags for nvcc based on CUDA_ARCH_NAME
+# Usage:
+#   SELECT_NVCC_ARCH_FLAGS(out_variable)
+#
+# Variables affecting the choice:
+#
+# CUDA_ARCH_NAME: One of ("Fermi" "Kepler" "Kepler-M" "Maxwell" "Maxwell-M" "All" "Manual")
+# ENV{CUDA_ARCH_BIN} : Only tested if CUDA_ARCH_NAME set to "Manual"
+#
+
+# Known NVIDIA GPU achitectures
+# This list will be used for CUDA_ARCH_NAME = All option
+
+set(CUDA_KNOWN_GPU_ARCHITECTURES "2.0 2.1(2.0) 3.0 3.5 5.0")
```

Comment:
Any reason this is a string instead of a list:
set(CUDA_KNOWN_GPU_ARCHITECTURES 2.0 2.1(2.0) 3.0 3.5 5.0)
...
list(APPEND CUDA_KNOWN_GPU_ARCHITECTURES 6.0 6.2)